staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime_test.go

```diff
@@ -69,3 +74,64 @@ func TestCustomHandleError(t *testing.T) {
 		t.Errorf("did not receive custom handler")
 	}
 }
+
+func TestHandleCrashLog(t *testing.T) {
+	log, err := captureStderr(func() {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Fatalf("expected a panic to recover from")
+			}
+		}()
+		defer HandleCrash()
+		panic("test panic")
+	})
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	// Example log:
+	//
+	// ...] Observed a panic: test panic
+	// goroutine 6 [running]:
+	// command-line-arguments.logPanic(0x..., 0x...)
+	// 	.../src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:69 +0x...
+	lines := strings.Split(log, "\n")
+	if len(lines) < 4 {
+		t.Fatalf("panic log should have 1 line of message, 1 line per goroutine and 2 lines per function call")
+	}
+	if match, _ := regexp.MatchString("Observed a panic: test panic", lines[0]); !match {
+		t.Errorf("mismatch panic message: %s", lines[0])
+	}
+	if match, _ := regexp.MatchString(`goroutine [0-9]+ \[running\]:`, lines[1]); !match {
+		t.Errorf("mismatch goroutine: %s", lines[1])
+	}
+	if match, _ := regexp.MatchString(`logPanic(.*)`, lines[2]); !match {
+		t.Errorf("mismatch symbolized function name: %s", lines[2])
+	}
+	if match, _ := regexp.MatchString(`runtime\.go:69 \+0x`, lines[3]); !match {
```

Comment:
this seems like a fragile test... is this format and line number stable across releases?

Reply:
same question for the `[running]` format, etc

Reply:
line #: it's fragile. I meant to have people update the number when changing our `runtime.go`, but it's not necessary

`[running]`: it comes from one of [the statuses](https://github.com/golang/go/blob/ea9859f858b603cbf49f0a930f83c56a716490a4/src/runtime/traceback.go#L879-L889). Should be stable in this test

updated both to use regexp. The rest of the test should be stable enough

also added documentation about if golang stdlib changes its format